core/src/main/java/org/mule/runtime/core/privileged/processor/CompositeProcessorChainRouter.java

```diff
@@ -76,6 +78,10 @@ protected ReactiveProcessor getExecutableFunction() {
     };
   }
 
+  protected List<? extends ReactiveProcessor> processorChainsToExecute() {
```

Comment:
make this method receive the chains by param instead of querying the internal state of this object